spec/lib/food_robot_spec.rb

```diff
@@ -0,0 +1,294 @@
+require 'rails_helper'
+
+RSpec.describe FoodRobot do
+  let(:monday) { Time.zone.today.monday }
+  let(:tuesday) { monday + 1.day }
+
+  describe '::generate_log_entries' do
+    context 'without an absence' do
+      context 'with an irregular schedule chain' do
+        let!(:chain) { create(:schedule_chain, day_of_week: monday.wday, irregular: true) }
+        let!(:donor) { create(:donation_schedule, schedule_chain: chain, position: 1) }
+        let!(:recipient) { create(:recipient_schedule, schedule_chain: chain, position: 2) }
+
+        subject { described_class.generate_log_entries(monday) }
+
+        it 'generates no logs' do
+          expect { subject }.not_to change { Log.count }
+        end
+      end
+
+      context 'with a non-functional schedule chain' do
```

Comment:
An empty chain would not be "functional", based on the definition [here](https://github.com/boulder-food-rescue/food-rescue-robot/blob/master/app/models/schedule_chain.rb#L55-L59).

Since `generate_log_entries` uses `ScheduleChain#functional?`, I don't see too much value in testing different kinds of "non-functional" schedule chains (at least for now).